packages/components/drag-handle/src/DragHandle.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { forwardRef, useCallback, useState } from 'react';
+import { css, cx } from 'emotion';
+import { usePrimitive } from '@contentful/f36-core';
+import type { CommonProps } from '@contentful/f36-core';
+import { Drag } from '@contentful/f36-icons';
+import tokens from '@contentful/f36-tokens';
+
+import { styles } from './DragHandle.styles';
+
+const generateStyles = ({ isActive, isFocused, isHovered }) => {
+  if (isActive || isFocused || isHovered) {
+    return css({
+      backgroundColor: tokens.colorElementLight,
+      cursor: isActive ? 'grabbing' : 'grab',
+    });
+  }
+};
+
+export type DragHandleInternalProps = {
+  /**
+   * Applies styling for when the component is actively being dragged by
+   * the user
+   */
+  isActive?: boolean;
+  /**
+   * Applies focus styling
+   */
+  isFocused?: boolean;
+  /**
+   * Applies hover styling
+   */
+  isHovered?: boolean;
+  /**
+   * Label rendered in DragHandle - not visible on screen as its purpose
+   * is for screen readers only
+   */
+  label: string;
+};
+
+export type DragHandleProps = CommonProps & DragHandleInternalProps;
```

Comment:
Does this need to be an interface with `extend HTMLProps<ElementType> {...`?

Reply:
Do you think there's anything gained from being able to set common HTML props on the drag handle? The only thing I can think of is aria properties, but ideally we should be setting those ourselves.

Reply:
yeah typescript won't recognise all the compatible `{...otherProps}` that we didn't define, like `onClick` for example. 
I think this has to be the standard way we define interfaces for components, we had multiple issues opened because some props aren't supported like `title`, `onBlur` events, etc. I saw people adding `//ts-ignore` as well in consumer apps 😅

Reply:
> I saw people adding //ts-ignore as well in consumer apps 😅

Ouch, that's problematic. I agree with you, but I also don't think we should do it everywhere without thinking it through, there are likely cases where we don't want to support custom event handlers and such.

Reply:
This should work now with the polymorphic props.

Reply:
Yes, It does work with the polymorphic component. good to know 👍🏼